### ISSUE TITLE
`edge-js` is looking for "Microsoft.CSharp', version: '4.5.0'" despite having set `process.env.EDGE_USE_CORECLR = '1';`

### ISSUE THREAD

**Issue by brianjenkins94** [agracio/edge-js#96]
**Sample to reproduce the issue:**

```
$ npm install edge-js
$ node -e "process.env.EDGE_USE_CORECLR = '1'; require('edge-js').func('async (input) => { return input.ToString(); }')"
Error: assembly specified in the dependencies manifest was not found -- package: 'Microsoft.CSharp', version: '4.5.0', path: 'lib/netstandard1.3/Microsoft.CSharp.dll'
/Users/bjenks/Desktop/temp/node_modules/edge-js/lib/edge.js:182
    return edge.initializeClrFunc(options);
                ^

TypeError: edge.initializeClrFunc is not a function
    at Object.exports.func (/Users/bjenks/Desktop/temp/node_modules/edge-js/lib/edge.js:182:17)
    at [eval]:1:56
    at Script.runInThisContext (vm.js:116:20)
    at Object.runInThisContext (vm.js:306:38)
    at Object.<anonymous> ([eval]-wrapper:9:26)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at evalScript (internal/process/execution.js:80:25)
    at internal/main/eval_string.js:23:3
```

**Versions:**

```
$ defaults read loginwindow SystemVersionStampAsString
10.15.2
$ node -v
v12.12.0
$ npm -v
6.13.2
$ dotnet --version
3.1.100
```

Any ideas?

**Comment by brianjenkins94** [comment]
I may have gotten past this error and on to [this error](https://github.com/agracio/edge-js-quick-start/issues/5).

**Comment by brianjenkins94** [comment]
<img width="1440" alt="Screen Shot 2019-12-21 at 12 59 01 PM" src="https://user-images.githubusercontent.com/9398474/71311832-b80a6180-23f1-11ea-8fc7-d430e39cc7e0.png">

Updated everything to use `dotnetcore3.1`, got it to build successfully in both Visual Studio and from the terminal, but now:

```
CoreClrEmbedding::Initialize - Failed to initialize CoreCLR, HRESULT: 0x80070057
/Users/bjenks/Developer/node-native-addon-experiments/edge/edge-js/lib/edge.js:145
    return edge.initializeClrFunc(options);
                ^

TypeError: edge.initializeClrFunc is not a function
```

**Comment by brianjenkins94** [comment]
Attaching [sampleThatReproducesTheIssue.zip](https://github.com/agracio/edge-js/files/3991686/sampleThatReproducesTheIssue.zip) which is a quite gutted copy of the project with everything but the CoreCLR removed.

**Comment by brianjenkins94** [comment]
Managed to get here by updating the paths in `edge.js` to point to `netstandard3.1`:

```
Load edge native library from: /Users/bjenks/Developer/node-native-addon-experiments/edge/edge-js/build/Release/edge_coreclr
Error: assembly specified in the dependencies manifest was not found -- package: 'Microsoft.CodeAnalysis.CSharp.Workspaces', version: '3.4.0', path: 'lib/netstandard2.0/Microsoft.CodeAnalysis.CSharp.Workspaces.dll'
/Users/bjenks/Developer/node-native-addon-experiments/edge/edge-js/lib/edge.js:145
    return edge.initializeClrFunc(options);
                ^

TypeError: edge.initializeClrFunc is not a function
    at Object.exports.func (/Users/bjenks/Developer/node-native-addon-experiments/edge/edge-js/lib/edge.js:145:17)
    at [eval]:1:62
    at Script.runInThisContext (vm.js:116:20)
    at Object.runInThisContext (vm.js:306:38)
    at Object.<anonymous> ([eval]-wrapper:9:26)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at evalScript (internal/process/execution.js:80:25)
    at internal/main/eval_string.js:23:3
```

**Comment by brianjenkins94** [comment]
Strangely not fixed by explicit reference:

```
Error: assembly specified in the dependencies manifest was not found -- package: 'Microsoft.CodeAnalysis.CSharp.Workspaces', version: '3.4.0', path: '/Users/bjenks/.nuget/packages/microsoft.codeanalysis.csharp.workspaces/3.4.0/lib/netstandard2.0/Microsoft.CodeAnalysis.CSharp.Workspaces.dll'
/Users/bjenks/Developer/node-native-addon-experiments/edge/edge-js/lib/edge.js:145
    return edge.initializeClrFunc(options);
                ^

TypeError: edge.initializeClrFunc is not a function
    at Object.exports.func (/Users/bjenks/Developer/node-native-addon-experiments/edge/edge-js/lib/edge.js:145:17)
    at [eval]:1:62
    at Script.runInThisContext (vm.js:116:20)
    at Object.runInThisContext (vm.js:306:38)
    at Object.<anonymous> ([eval]-wrapper:9:26)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at evalScript (internal/process/execution.js:80:25)
    at internal/main/eval_string.js:23:3
```

**Comment by corylulu** [comment]
This is old, but I think I found the problem. After testing on 3 different systems and not being able to get it to initialize properly on Linux, I found that in **deps_entry.cpp**, the `EDGE_PLATFORM_NIX` was not being defined, and because of that, the nuget packages path resolutions weren't being lower cased. 

I was also having a bad hash check issue, but that might be because of windows and using WSL/Docker for that stuff... (since they append files with permissions, I won't say that's related)

But the bad casing on the paths was the main problem I was having and shit change fixed it. 
https://github.com/agracio/edge-js/pull/101
